Limited number of days for weather report

AtreidesZero opened the pull request:
I limited the number of days for the weather report to the 2nd and 3rd entries in the list. Based on the time of night the bot normally runs this should limit the displayed information to the current day and current night. Skipping Tonight and the rest of the forecast.

Tweaking line 30 changes how many forecast entries are shown.

Newtons3Laws committed:
Limited number of days for weather report

aguynamedjoetoo commented on daily_post.py:
```diff
@@ -26,8 +26,10 @@ def get_weather():
-    forecast = weather_info.contents[0].text
-    reddit_comment = '* ' + str.replace(forecast, '\n\n',"\n* ")[:-3]
+    forecast = str(weather_info.contents[0]).split("<br/>\n<br/>")
+    top_three = '<br/>\n<br/>'.join(str(x) for x in forecast[1:3])
```
The major problem here is that on occasion, the forecast will have a "special update" at the top where the just arbitrarily throw it in :| It would be easy to catch if they used a constant tag or something, but they don't.

So in the event of a special update, this would return just the special update, and end up cutting off part of the day's weather.

AtreidesZero replied:
Any chance you have a sample of the api response when it includes a special update?

aguynamedjoetoo replied:
I don't have a specific BS4 response, unfortunately, but here are a few posts that have been caused from it:

https://www.reddit.com/r/SeattleWA/comments/7e7dlw/seattle_reddit_community_open_chat_monday/

https://www.reddit.com/r/SeattleWA/comments/7epce5/seattle_reddit_community_open_chat_wednesday/

https://www.reddit.com/r/SeattleWA/comments/7fdxkh/seattle_reddit_community_open_chat_saturday/

I'm wondering if we can at least get a list of known curveballs like this and just add a really dumb if block for formatting it if it comes in haha

AtreidesZero replied:
I managed to find some live versions.
[Example-Air-Stagnation-Advisory.txt](https://github.com/r-seattle-wa/SeattleRedditBot/files/1529719/Example-Air-Stagnation-Advisory.txt)
[Example-Hazardous-Weather-Outlook.txt](https://github.com/r-seattle-wa/SeattleRedditBot/files/1529720/Example-Hazardous-Weather-Outlook.txt)

I'll see if I can find if there are any examples. It's wrapped in a div and anchor tag so we can target it for removal if we want or scrap the link if we want.

AtreidesZero replied:
[Example-Double-Advisories.txt](https://github.com/r-seattle-wa/SeattleRedditBot/files/1529728/Example-Double-Advisories.txt)

Found one. I'll update the splicer section to handle these as their own thing.

Newtons3Laws committed:
Updated splicer to handle weather advisories, cleaned up some whitespace that made pycharm complain.
Updated template to reflect the reduced days in weather forecast. Changed days kept from 1 to 2 (line 37)

AtreidesZero commented:
As per our discussion I've added handling for weather advisories. If there are none no header is displayed for them. If one or more are found then it adds an "* Advisories: " line and adds each advisory found. I tried to make the links work but the links they generate are meant for an embedded widget and wouldn't work from reddit.

Then it skips the first entry in the list as the bot generally runs during the night ad the first entry is "Tonight" or "Overnight" and then picks up the next 4 records for a 2 day forecast.

aguynamedjoetoo commented on daily_post.py:
```diff
@@ -21,25 +21,35 @@
+        advisories = weather_info.div.extract().find_all('a')
+        reddit_comment += '* Advisories: \n'
+        for advis in advisories:
+            reddit_comment += ' * ' + advis.span.string + ' \n'
```
Extra space in front of the `*`?

Additionally it's a bit cleaner to use string formatting in instances like this:

`'* {}\n'.format(advis.span.string)`

AtreidesZero replied:
The space in front of the * makes it a nested bulleted list.  I'll update to your recommended formatting and open under a new PR.

aguynamedjoetoo replied:
Ah, yeah. Makes enough sense then.

Also it may just open up under this same PR haha if it does no sweat m8

aguynamedjoetoo commented on daily_post.py:
```diff
@@ -21,25 +21,35 @@
+    reddit_comment = ''
+    if len(weather_info.find_all('div')) > 0:
+        advisories = weather_info.div.extract().find_all('a')
+        reddit_comment += '* Advisories: \n'
```
OCD pickiness on my front, no need for the space after the `:`

aguynamedjoetoo replied:
Oh also, you can probably also just do `if weather_info.find_all('div')` since it would return null if nothing was found I believe.

aguynamedjoetoo commented on daily_post.py:
```diff
@@ -21,25 +21,35 @@
-    reddit_comment = '* ' + str.replace(forecast, '\n\n',"\n* ")[:-3]
+    weather_info = soup_doc.find_all('table')[1].contents[0].contents[0]
+    reddit_comment = ''
+    if len(weather_info.find_all('div')) > 0:
```
Oh boy, this is a pretty big assumption, but we can see if it bites us in the ass at all. Let's just keep an eye on it I suppose.
